FilePlanDescriptors for RetentionLabels

sseth-msft opened the pull request.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "appliedCategory resource type"
```
This complex type appears as `filePlanAppliedCategory` in the api.md file. Please review.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "appliedCategory resource type"
+description: "Represents the unique category applied to a retention label."
```
```suggestion
description: "Represents a unique category applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String|Unique string that defines a category name applied to the label. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
+|subCategory|[microsoft.graph.security.subCategory](../resources/security-fileplandescriptor-subcategory.md)|**TODO: Add Description**|
```
Please add a description.

sseth-msft replied:
Represents the unique subcategory under this category that is applied to a retention label.

Added this in the new PR.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authority.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "authority resource type"
+description: "Represents the unique authority applied to a retention label."
```
```suggestion
description: "Represents a unique authority applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authority.md:
```diff
@@ -0,0 +1,42 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the unique authority applied to a retention label.
```
```suggestion
Represents a unique authority applied to a retention label.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authority.md:
```diff
@@ -0,0 +1,42 @@
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String|Unique string that defines a authority name. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
```
```suggestion
|displayName|String|Unique string that defines an authority name. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+---
+title: "authorityTemplate resource type"
+description: "**Represents the 'authority' file plan descriptor for retention labels."
```
```suggestion
description: "Represents the authority file plan descriptor for retention labels."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the 'authority' file plan descriptor for retention labels.
```
```suggestion
Represents the authority file plan descriptor for retention labels.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+---
+title: "authorityTemplate resource type"
+description: "Represents the authority file plan descriptor for retention labels."
```
What is the purpose of the authority template? We have 4 nouns together, and it might confuse users. Perhaps, we can come up with a better description.

Danielabom replied:
```suggestion
description: "Represents the authority file plan descriptor for retention labels."
```
```suggestion
description: "Represents the file plan descriptor of type authority for retention labels."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the authority file plan descriptor for retention labels.
```
What is the purpose of the authority template? We have 4 nouns together, and it might confuse users. Perhaps, we can come up with a better description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+Inherits from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).
+
+## Methods
+|Method|Return type|Description|
```
I wanted to confirm that we have the correct methods here, but I did not find them in the api.md file.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+|:---|:---|:---|
+|createdBy|[microsoft.graph.identitySet](/graph/api/resources/identityset)|Represents the user who created the authority. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
+|createdDateTime|DateTimeOffset|Represents the date and time in which the authority is created. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
+|displayName|String|Unique string that defines a authority name. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
```
```suggestion
|displayName|String|Unique string that defines an authority name. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: "categoryTemplate resource type"
+description: "Represents the 'category' file plan descriptor for retention labels."
```
```suggestion
description: "Represents the category file plan descriptor for retention labels."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the 'category' file plan descriptor for retention labels.
```
```suggestion
Represents the category file plan descriptor for retention labels.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the category file plan descriptor for retention labels.
```
What is the purpose of the categoryTemplate? We have 4 nouns together, and it might confuse users. Perhaps, we can come up with a better description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: "categoryTemplate resource type"
+description: "Represents the category file plan descriptor for retention labels."
```
What is the purpose of the categoryTemplate? We have 4 nouns together, and it might confuse users. Perhaps, we can come up with a better description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+Inherits from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).
+
+## Methods
+|Method|Return type|Description|
```
Since the filenames of the resources are incorrect, some method names need to be corrected as well.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
```
The filenames are incorrect, please review the following wiki: (https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211759/OData-subnamespace)

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+Inherits from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).
+
+## Methods
+|Method|Return type|Description|
```
Since the filenames of the resources are incorrect, some method names need to be corrected as well.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+## Relationships
+|Relationship|Type|Description|
+|:---|:---|:---|
+|subCategories|[microsoft.graph.security.subCategoryTemplate](../resources/security-fileplandescriptor-subcategorytemplate.md) collection|**TODO: Add Description**|
```
Please add a valid description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citation.md:
```diff
@@ -0,0 +1,46 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|citationJurisdiction|String|Represents the user who created the citation.|
+|citationUrl|String|**TODO: Add Description**|
```
Please add a valid description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citation.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "citation resource type"
+description: "Represents the unique citation applied to a retention label."
```
```suggestion
description: "Represents a unique citation applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: "citationTemplate resource type"
+description: "Represents the 'citation' file plan descriptor for retention labels."
```
Please apply the same changes reflected in previous resource descriptions.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the 'citation' file plan descriptor for retention labels.
```
Please apply the same changes reflected in previous resource descriptions.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Global comment to all the resources with methods: I wanted to confirm that we have the correct methods here, but I did not find them in the api.md file.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|citationJurisdiction|String|**TODO: Add Description**|
```
Please add a valid description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|citationJurisdiction|String|**TODO: Add Description**|
+|citationUrl|String|**TODO: Add Description**|
```
Please add a valid description.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-department.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "department resource type"
+description: "Represents the unique department applied to a retention label."
```
```suggestion
description: "Represents a unique department applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-department.md:
```diff
@@ -0,0 +1,42 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the unique department applied to a retention label.
```
```suggestion
Represents a unique department applied to a retention label.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: "departmentTemplate resource type"
+description: "Represents the 'department' file plan descriptor for retention labels."
```
Please apply the same changes reflected in previous resource descriptions.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the 'department' file plan descriptor for retention labels.
```
Please apply the same changes reflected in previous resource descriptions.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Since the filenames of the resources are incorrect, some method names need to be corrected as well.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-fileplanreference.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "filePlanReference resource type"
+description: "Represents the unique filePlanReference applied to a retention label."
```
```suggestion
description: "Represents a unique filePlanReference applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor-fileplanreference.md:
```diff
@@ -0,0 +1,42 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the unique filePlanReference applied to a retention label.
```
```suggestion
Represents a unique filePlanReference applied to a retention label.
```

Danielabom reviewed (changes requested):
@sseth-msft, thank you for working on this PR. The PR did not meet our first review requirements.
Please note that:
The filenames of the resources are incorrect, resulting in incorrect changelog links, and some methods' names are likely to be incorrect. 
I found many empty descriptions in the files; please add valid descriptions.
Most of the comments in this PR are global, meaning that the comments posted apply to all the resources in this PR. 
I decided not to review the methods until their names are updated. Thanks.

Danielabom commented on api-reference/beta/resources/security-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+---
+title: "authorityTemplate resource type"
+description: "Read the properties and relationships of a microsoft.graph.security.authorityTemplate object."
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the authorityTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it in here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md) object.
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the authorityTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations.

Danielabom commented on api-reference/beta/resources/security-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Since the names of the methods need to be corrected, the links to the operations listed below must be updated.

Danielabom commented on api-reference/beta/resources/security-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: "categoryTemplate resource type"
+description: "Read the properties and relationships of a microsoft.graph.security.categoryTemplate object."
```
Perhaps, I am confused, but this sounds like the description of a method instead of a resource. What is the categoryTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md) object.
```
Perhaps, I am confused, but this sounds like the description of a method instead of a resource. What is the categoryTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations.

Danielabom commented on api-reference/beta/resources/security-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Since the names of the methods need to be corrected, the links of the operations listed below must be updated.

Danielabom commented on api-reference/beta/resources/security-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+## Relationships
+|Relationship|Type|Description|
+|:---|:---|:---|
+|subCategories|[microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) collection|**TODO: Add Description**|
```
Global comment: Please add the missing descriptions.

Danielabom commented on api-reference/beta/resources/security-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+---
+title: "citationTemplate resource type"
+description: "Read the properties and relationships of a microsoft.graph.security.citationTemplate object."
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the citationTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.citationTemplate](../resources/security-citationtemplate.md) object.
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the citationTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations.

Danielabom commented on api-reference/beta/resources/security-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Since the names of the methods need to be corrected, the links of the operations listed below must be updated.

Danielabom commented on api-reference/beta/resources/security-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: "departmentTemplate resource type"
+description: "Read the properties and relationships of a microsoft.graph.security.departmentTemplate object."
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the departmentTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.departmentTemplate](../resources/security-departmenttemplate.md) object.
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the departmentTemplate resource? What does it represent? Or what does it provide?

Danielabom commented on api-reference/beta/resources/security-departmenttemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
```
Since the names of the methods need to be corrected, the links of the operations listed below must be updated.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+---
+title: "filePlanDescriptor resource type"
+description: "**TODO: Add Description**"
```
Please add valid descriptions.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and the retention schedule for it. |
```
```suggestion
|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and its retention schedule. |
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and the retention schedule for it. |
+|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a certain department.|
```
```suggestion
|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a particular department.|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and the retention schedule for it. |
+|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a certain department.|
+|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|Specifies the unique identification defining a rule or regulation, following which, the label and content to which it is applied is scheduled to be retained or to be deleted. Additionally, you can specify the direct url of the citation and the jurisdiction or agency which published it.|
```
This description needs to be updated; it might need to be clarified for readers.

sseth-msft replied:
Specifies the unique rule or regulation published by a jurisdiction, following which, the label and content to which it is applied is scheduled to be retained or to be deleted. Additionally, you can specify a direct link to the citation and the jurisdiction or agency which published it.
@Danielabom does this sound better?

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and the retention schedule for it. |
+|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a certain department.|
+|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|Specifies the unique identification defining a rule or regulation, following which, the label and content to which it is applied is scheduled to be retained or to be deleted. Additionally, you can specify the direct url of the citation and the jurisdiction or agency which published it.|
+|department|[microsoft.graph.security.department](../resources/security-department.md)|Specifies the parent business vertical or department of an organization for a label and in turn the content to which it will be applied.|
```
This description needs to be updated; it might need to be clarified for readers.

sseth-msft replied:
@Danielabom - any suggestions on this?

Danielabom replied:
A label assigned to content based on the department or business unit it belongs to within an organization

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a certain department.|
+|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|Specifies the unique identification defining a rule or regulation, following which, the label and content to which it is applied is scheduled to be retained or to be deleted. Additionally, you can specify the direct url of the citation and the jurisdiction or agency which published it.|
+|department|[microsoft.graph.security.department](../resources/security-department.md)|Specifies the parent business vertical or department of an organization for a label and in turn the content to which it will be applied.|
+|filePlanReference|[microsoft.graph.security.filePlanReference](../resources/security-fileplanreference.md)|Specifies a unique identifier of a label category which co-relates it to the that organization’s retention schedule. It is an alpha-numeric identifier that maps the retention label to a row on their retention schedule. |
```
This description needs to be updated; it might need to be clarified for readers.

sseth-msft replied:
modified the desc

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+## Relationships
+|Relationship|Type|Description|
+|:---|:---|:---|
+|authorityTemplate|[microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md)|Represent the properties of 'authority' file plan descriptor.|
```
I cannot comment on the property until the description of the authorityTemplate file is updated.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|Relationship|Type|Description|
+|:---|:---|:---|
+|authorityTemplate|[microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md)|Represent the properties of 'authority' file plan descriptor.|
+|categoryTemplate|[microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md)|Represent the properties of 'category' file plan descriptor.|
```
I cannot comment on the property until the description of the categoryTemplate file is updated.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|:---|:---|:---|
+|authorityTemplate|[microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md)|Represent the properties of 'authority' file plan descriptor.|
+|categoryTemplate|[microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md)|Represent the properties of 'category' file plan descriptor.|
+|citationTemplate|[microsoft.graph.security.citationTemplate](../resources/security-citationtemplate.md)|Represent the properties of 'citation' file plan descriptor.|
```
```suggestion
I cannot comment on the property until the description of the citationTemplate file is updated.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|authorityTemplate|[microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md)|Represent the properties of 'authority' file plan descriptor.|
+|categoryTemplate|[microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md)|Represent the properties of 'category' file plan descriptor.|
+|citationTemplate|[microsoft.graph.security.citationTemplate](../resources/security-citationtemplate.md)|Represent the properties of 'citation' file plan descriptor.|
+|departmentTemplate|[microsoft.graph.security.departmentTemplate](../resources/security-departmenttemplate.md)|Represent the properties of 'department' file plan descriptor.|
```
I cannot comment on the property until the description of the departmentTemplate file is updated.


```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|categoryTemplate|[microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md)|Represent the properties of 'category' file plan descriptor.|
+|citationTemplate|[microsoft.graph.security.citationTemplate](../resources/security-citationtemplate.md)|Represent the properties of 'citation' file plan descriptor.|
+|departmentTemplate|[microsoft.graph.security.departmentTemplate](../resources/security-departmenttemplate.md)|Represent the properties of 'department' file plan descriptor.|
+|filePlanReferenceTemplate|[microsoft.graph.security.filePlanReferenceTemplate](../resources/security-fileplanreferencetemplate.md)|Represent the properties of 'file plan reference identification' file plan descriptor.|
```
I cannot comment on the property until the description of the filePlanReferenceTemplate file is updated.

Danielabom commented on api-reference/beta/resources/security-fileplanreferencetemplate.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: "filePlanReferenceTemplate resource type"
+description: "Read the properties and relationships of a microsoft.graph.security.filePlanReferenceTemplate object."
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the filePlanReferenceTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-fileplanreferencetemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.filePlanReferenceTemplate](../resources/security-fileplanreferencetemplate.md) object.
```
Perhaps, I am confused but this sounds like the description of a method instead of a resource. What is the filePlanReferenceTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-subcategory.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "subCategory resource type"
+description: "Represents the unique subcategory applied to a retention label."
```
```suggestion
description: "Represents a unique subcategory applied to a retention label."
```

Danielabom commented on api-reference/beta/resources/security-subcategory.md:
```diff
@@ -0,0 +1,42 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the unique subcategory applied to a retention label.
```
```suggestion
Represents a unique subcategory applied to a retention label.
```

Danielabom commented on api-reference/beta/resources/security-subcategorytemplate.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: "subCategoryTemplate resource type"
+description: "Read the properties and relationships of a [microsoft.graph.security.subcategoryTemplate object."
```
Perhaps, I am confused, but this sounds like the description of a method instead of a resource. What is the subCategoryTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations. Also, I believe you attempted to add the link to the resource, but we do not need it here. We add in the second description, which is line 16.

Danielabom commented on api-reference/beta/resources/security-subcategorytemplate.md:
```diff
@@ -0,0 +1,61 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.subcategoryTemplate](../resources/security-subcategorytemplate.md) object.
```
Perhaps, I am confused, but this sounds like the description of a method instead of a resource. What is the subCategoryTemplate resource? What does it represent? Or what does it provide? We normally use the verb read to describe "Get" operations.

Danielabom reviewed (changes requested):
Hi Sam, 
Thank you for updating the files on the PR.
I am concerned about feedback not being addressed in the first round, which may create confusion and prevent us from completing the review. Previously, I mentioned that we need full descriptions to approve or enhance them together. Having "TODO: Add description" phrases as placeholders would typically be a reason to consider that a PR is not ready for review. 

I could not review the TOC because I need to make we are placing the correct methods under the correct parent entities. 

Regarding the methods' names:
Please review the file naming wiki previously shared (https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211751/File-naming), and update the file names that do not follow the guidelines. I commented on one of the methods with an example that applies to that specific operation.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Specifies the underlying authority that describes the type of content to be retained and its retention schedule. |
+|category|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Specifies a group of similar types of content in a particular department.|
+|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|Specifies the unique rule or regulation published by a jurisdiction, following which, the label and content to which it is applied is scheduled to be retained or to be deleted. Additionally, you can specify a direct link to the citation and the jurisdiction or agency which published it.|
```
```suggestion
|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|The specific rule or regulation created by a jurisdiction used to determine whether certain labels and content should be retained or deleted. 
```

Danielabom replied:
@sseth-msft, would it be valuable to update the "citation" complex type description, too?

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
```
@sseth-msft  since we are updating the properties' descriptions and these at the same are either entities or complex types. Consider updating the descriptions in the corresponding files. Let me know if you have questions.

sseth-msft replied:
done

sseth-msft commented:
> Hi Sam, Thank you for updating the files on the PR. I am concerned about feedback not being addressed in the first round, which may create confusion and prevent us from completing the review. Previously, I mentioned that we need full descriptions to approve or enhance them together. Having "TODO: Add description" phrases as placeholders would typically be a reason to consider that a PR is not ready for review.
> 
> I could not review the TOC because I need to make we are placing the correct methods under the correct parent entities.
> 
> Regarding the methods' names: Please review the file naming wiki previously shared (https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211751/File-naming), and update the file names that do not follow the guidelines. I commented on one of the methods with an example that applies to that specific operation.

Addressed all the feedbacks you've shared so far.

Danielabom commented on api-reference/beta/resources/security-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "appliedCategory resource type"
+description: "Represents the file plan descriptor of type category applied to a particular retention labels."
```
```suggestion
description: "Represents the file plan descriptor of type category applied to a particular retention labels."
```
```suggestion
description: "Represents the file plan descriptor of type category applied to a particular retention label."
```

Danielabom commented on api-reference/beta/resources/security-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String|Unique string that defines a category name applied to the label. Inherited from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).|
+|subCategory|[microsoft.graph.security.subCategory](../resources/security-subcategory.md)|Represents the file plan descriptor of type subcategory under this category that is applied to a particular  retention label.|
```
```suggestion
|subCategory|[microsoft.graph.security.subCategory](../resources/security-subcategory.md)|Represents the file plan descriptor for a subcategory under a specific category, which has been assigned to a particular retention label.|
```

Danielabom commented on api-reference/beta/resources/security-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the file plan descriptor of type category applied to a particular retention labels.
```
```suggestion
Represents the file plan descriptor of type category applied to a particular retention labels.
```
```suggestion
Represents the file plan descriptor of the type category applied to a particular retention label.
```

Danielabom commented on api-reference/beta/resources/security-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "appliedCategory resource type"
+description: "Represents the file plan descriptor of type category applied to a particular retention label."
```
```suggestion
description: "Represents the file plan descriptor of type category applied to a particular retention label."
```
```suggestion
description: "Represents the file plan descriptor of the type category applied to a particular retention label."
```

Danielabom commented on api-reference/beta/resources/security-appliedcategory.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "appliedCategory resource type"
+description: "Represents the file plan descriptor of the type category applied to a particular retention label."
```
```suggestion
description: "Represents the file plan descriptor of type category applied to a particular retention label."
```

Danielabom commented on api-reference/beta/api/security-authoritytemplate-list.md:
```diff
@@ -0,0 +1,92 @@
+---
+title: "List authorityTemplates"
+description: "Get a list of the microsoft.graph.security.authorityTemplate objects and their properties."
```
```suggestion
description: "Get a list of the microsoft.graph.security.authorityTemplate objects and their properties."
```

Danielabom replied:
**As mentioned before,** this seems to be an attempt to include the link. Descriptions in line 3 do not have links. Please use the correct format of the resource name here, example: 
Instead of microsoft.graph.security.authorityTemplate, use authorityTemplate

Danielabom commented on api-reference/beta/api/security-authoritytemplate-get.md:
```diff
@@ -0,0 +1,91 @@
+---
+title: "Get authorityTemplate"
+description: "Read the properties and relationships of a microsoft.graph.security.authorityTemplate object."
```
_**GLOBAL COMMENT: As mentioned before**_, this seems to be an attempt to copy the description from line 15. Descriptions in line 3 do not have links. Please use the correct resource names. For example: 
Instead of using  microsoft.graph.security.authorityTemplate object, use .authorityTemplate

Danielabom commented on api-reference/beta/resources/security-categorytemplate.md:
```diff
@@ -0,0 +1,65 @@
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List categoryTemplates](../api/security-categorytemplate-list.md)|[microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md) collection|Get a list of the [microsoft.graph.security.categoryTemplate](../resources/security-categorytemplate.md) objects and their properties.|
```
_**GLOBAL COMMENT: As mentioned before,**_ by updating the method names, the links in the methods table of the resource files also needed to be updated.

Danielabom commented on api-reference/beta/api/security-categorytemplate-list-subcategories.md:
```diff
@@ -0,0 +1,92 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the subCategoryTemplate resources from the subCategories navigation property.
```
Please add the corresponding links in this description.

Danielabom commented on api-reference/beta/api/security-categorytemplate-list.md:
```diff
@@ -0,0 +1,92 @@
+---
+title: "List categoryTemplates"
+description: "Get a list of the microsoft.graph.security.categoryTemplate objects and their properties."
```
As mentioned before, please update this description format.

Danielabom commented on api-reference/beta/resources/security-citation.md:
```diff
@@ -0,0 +1,46 @@
+---
+title: "citation resource type"
+description: "Represents the file plan descriptor of type citation applied to a particular retention labels."
```
```suggestion
description: "Represents the file plan descriptor of type citation applied to a particular retention label."
```

Danielabom reviewed (changes requested):
Hi @sseth-msft, we are having issues with the method names. I am trying to give you the correct file names but we are missing files. Please add the labelsRoot file with the new navigation properties. I also added some Global comments, which are the last review's callouts. Please make sure to address them. Also, did you manually create these files?

sseth-msft commented:
> Hi @sseth-msft, we are having issues with the method names. I am trying to give you the correct file names but we are missing files. Please add the labelsRoot file with the new navigation properties. I also added some Global comments, which are the last review's callouts. Please make sure to address them. Also, did you manually create these files?

Can you please point me to where the labelsroot file is needed? Sometimes there are files with same content generated and the labelsroot was a duplicate that way. I have not created any new files myself, I may have deleted the ones that are not needed. 
I can address the issue with the right file if understand the context. Fixes all other comments regarding descriptions.

Danielabom reviewed (changes requested):
Sam, I can point you to the CSDL file you shared. Please look up the entity ¨labelsRoot¨. There, you'll find the nav properties that need to be documented in green. I suggest going through the entire CSDL and making sure that you've added all the files we need.

Danielabom commented on api-reference/beta/resources/security-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+he specific rule or regulation created by a jurisdiction used to determine whether certain labels and content should be retained or deleted.
```
```suggestion
The specific rule or regulation created by a jurisdiction used to determine whether certain labels and content should be retained or deleted.
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Represents the file plan descriptor of type authority applied to a particular retention label.|
+|appliedcategory|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Represents the file plan descriptor of type category applied to a particular retention labels.|
```
Your api.md and csdl show the name of this property as just "category".

Danielabom replied:
```suggestion
|appliedcategory|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Represents the file plan descriptor of type category applied to a particular retention label.|
```

Danielabom commented on api-reference/beta/resources/security-citationtemplate.md:
```diff
@@ -0,0 +1,65 @@
+|[Delete citationTemplate](../api/security-citationtemplate-delete.md)|None|Delete a [microsoft.graph.security.citationTemplate](../resources/security-citationtemplate.md) object.|
+
+## Properties
+|Property|Type|Description|
```
The api.md shows a property called "descriptor" not listed here. Please review.

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Represents the file plan descriptor of type authority applied to a particular retention label.|
+|appliedcategory|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Represents the file plan descriptor of type category applied to a particular retention labels.|
+|citation|[microsoft.graph.security.citation](../resources/security-citation.md)|Represents the file plan descriptor of type citation applied to a particular retention labels.|
+|department|[microsoft.graph.security.department](../resources/security-department.md)|Represents the file plan descriptor of type department applied to a particular retention labels.|
```
```suggestion
|department|[microsoft.graph.security.department](../resources/security-department.md)|Represents the file plan descriptor of type department applied to a particular retention label.|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,92 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|authority|[microsoft.graph.security.authority](../resources/security-authority.md)|Represents the file plan descriptor of type authority applied to a particular retention label.|
+|appliedcategory|[microsoft.graph.security.appliedCategory](../resources/security-appliedcategory.md)|Represents the file plan descriptor of type category applied to a particular retention label.|
```
Your api.md and csdl show the name of this property as just "category".

Danielabom commented on api-reference/beta/resources/security-fileplanreference.md:
```diff
@@ -0,0 +1,42 @@
+---
+title: "filePlanReference resource type"
+description: "Represents the file plan descriptor of type filePlanReference applied to a particular retention labels."
```
```suggestion
description: "Represents the file plan descriptor of type filePlanReference applied to a particular retention label."
```

Danielabom commented on api-reference/beta/resources/security-fileplanreference.md:
```diff
@@ -0,0 +1,42 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the file plan descriptor of type filePlanReference applied to a particular retention labels.
```
```suggestion
Represents the file plan descriptor of type filePlanReference applied to a particular retention label.
```

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+---
+title: "labelsRoot resource type"
+description: "Represents the entry point for retention labels resource type."
```
```suggestion
description: "Represents the entry point for the retentionLabels resource type."
```

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the entry point for retention labels resource type.
```
```suggestion
Represents the entry point for the retentionLabels resource type.
```

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|
```
Does the resource have properties, or should it be "none"? Please update the section accordingly.

Danielabom commented on api-reference/beta/resources/security-retentionlabel.md:
```diff
@@ -51,7 +54,10 @@ Represents how customers can manage their data, including whether and for how lo
 |:---|:---|:---|
 |dispositionReviewStages|[microsoft.graph.security.dispositionReviewStage](../resources/security-dispositionreviewstage.md) collection|When action at the end of retention is chosen as 'dispositionReview', dispositionReviewStages specifies a sequential set of stages with at least one reviewer in each stage.|
-|eventType|[microsoft.graph.security.retentionEventType](../resources/security-retentioneventtype.md)|Represents the type associated with a retention event..|
+|eventType|[microsoft.graph.security.retentionEventType](../resources/security-retentioneventtype.md)|Represents the type associated with a retention event.|
```
Based on the CSDL and the api.md files, the name of this nav property is "retentionEventType."

Danielabom reviewed (changes requested):
Regarding the new file names:
I've made all the changes related to file naming issues. Please be aware that all the file naming changes require changes from your side, like updating the methods' names and links (only those with new file names). 

Missing files:
As explained before, whatever appears as "new ..." in the CSDL and the api.md files needs to be documented in the PR.

In this case, we are missing:
An entity called filePlanDescriptorTemplate
A complex type called filePlanDescriptorBase 

In the changelog:
Add new methods (list and create) for retentionLabel
Add the new resource labelsRoot

In the api-reference/beta/resources/security-retentionlabel.md file:
According to the CSDL, "dispositionReviewStages" is not a property. It is a nav property, already documented in relationships.
Based on the CSDL and the api.md files, the name of this nav property is "retentionEventType."
FYi, the api.md file indicates that the dispositionReviewStage resource only supports a create operation, not an update operation. 

In the api-reference/beta/resources/security-fileplandescriptor.md file: 
FYI, the resource filePlanDescriptor appears as filePlanDescriptors in the api.md file dot MD file.
FYI, the api.md and csdl show the property “appliedcategory” as just "category".

In the api-reference/beta/resources/security-labelsroot.md file:
Does the resource have properties, or should it be "none"? Please update the section accordingly.

TOC
Now, that you have the correct file names, please update the TOC accordingly. Here's the link to the TOC guidelines: [ToC and topic title - Overview (azure.com)](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211766/ToC-and-topic-title)

angelgolfer-ms commented on api-reference/beta/resources/security-authoritytemplate.md:
```diff
@@ -0,0 +1,60 @@
+
+Specifies the underlying authority that describes the type of content to be retained and its retention schedule.
+
+Inherits from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).
```
Line 18 and similar lines for the other 5 resources that represent templates that are based on filePlanDescriptorTemplate resource:
Hi @sseth-msft ,
Just a quick note as your reviewer has asked me to look at a few things in this PR, and I happened to notice this - for all 6 resources derived from filePlanDescriptorTemplate as defined in your schema, this line should say "Inherits from microsoft.graph.security.filePlanDescriptorTemplate, not microsoft.graph.security.filePlanDescriptor.

Can you update all such 6 instances?

angelgolfer-ms replied:
Also, please document filePlanDescriptorTemplate.

sseth-msft replied:
Added the file and made the suggested changes in all 6 files.

angelgolfer-ms commented on api-reference/beta/resources/security-authority.md:
```diff
@@ -0,0 +1,42 @@
+Represents the file plan descriptor of type authority applied to a particular retention label.
+
+
+Inherits from [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md).
```
Line 19 and similar lines in the other 5 resources derived from filePlanDescriptorBase in your schema:
This line should say microsoft.graph.security.filePlanDescriptorBase, not microsoft.graph.security.filePlanDescriptor.

For that reason, you should also document filePlanDescriptorBase.

sseth-msft replied:
Made changes to all 6 files.

Danielabom commented on changelog/Microsoft.Security.RecordsManagement.json:
```diff
@@ -218,6 +218,150 @@
+            "Target": "labelsRoot"
+          },{
+            "Id": "523c6227-aa1d-473f-8c16-50fd37ca12c6",
+            "ApiChange": "LIST",
```
```suggestion
            "ApiChange": "GET",
```

Danielabom commented on changelog/Microsoft.Security.RecordsManagement.json:
```diff
@@ -218,6 +218,150 @@
+            "ApiChange": "LIST",
+            "ChangedApiName": "retentionLabel",
+            "ChangeType": "Addition",
+            "Description": "Added the [list filePlanDescriptor](https://learn.microsoft.com/en-us/graph/api/security-retentionlabel-list?view=graph-rest-beta) operation to [retentionLabel](https://learn.microsoft.com/en-us/graph/api/resources/security-retentionLabel?view=graph-rest-beta) resource",
```
```suggestion
            "Description": "Added the [list]](https://learn.microsoft.com/en-us/graph/api/security-retentionlabel-list?view=graph-rest-beta) operation to [retentionLabel](https://learn.microsoft.com/en-us/graph/api/resources/security-retentionLabel?view=graph-rest-beta) resource",
```

Danielabom commented on changelog/Microsoft.Security.RecordsManagement.json:
```diff
@@ -218,6 +218,150 @@
+            "ApiChange": "POST",
+            "ChangedApiName": "retentionLabel",
+            "ChangeType": "Addition",
+            "Description": "Added the [post filePlanDescriptor](https://learn.microsoft.com/en-us/graph/api/security-retentionlabel-post?view=graph-rest-beta) operation to [retentionLabel](https://learn.microsoft.com/en-us/graph/api/resources/security-retentionLabel?view=graph-rest-beta) resource",
```
```suggestion
            "Description": "Added the [post] (https://learn.microsoft.com/en-us/graph/api/security-retentionlabel-post?view=graph-rest-beta) operation to [retentionLabel](https://learn.microsoft.com/en-us/graph/api/resources/security-retentionLabel?view=graph-rest-beta) resource",
```

Danielabom commented on api-reference/beta/resources/security-retentionlabel.md:
```diff
@@ -23,9 +23,12 @@ Represents how customers can manage their data, including whether and for how lo
 |[Update retentionLabel](../api/security-retentionlabel-update.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Update the properties of a [retentionLabel](../resources/security-retentionlabel.md) object.|
 |[Delete retentionLabel](../api/security-retentionlabel-delete.md)|None|Delete a [retentionLabel](../resources/security-retentionlabel.md) object.|
+|[List filePlanDescriptor](../api/security-retentionlabel-list.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get the filePlanDescriptor resources from the descriptors navigation property.|
+|[Create filePlanDescriptor](../api/security-retentionlabel-post.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md)|Create a new filePlanDescriptor object.|
```
```suggestion
|[Create filePlanDescriptor](../api/security-labelsroot-post-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md)|Create a new filePlanDescriptor object.|
```

Danielabom commented on api-reference/beta/resources/security-retentionlabel.md:
```diff
@@ -23,9 +23,12 @@ Represents how customers can manage their data, including whether and for how lo
 |[Get retentionLabel](../api/security-retentionlabel-get.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Read the properties and relationships of a [retentionLabel](../resources/security-retentionlabel.md) object.|
 |[Update retentionLabel](../api/security-retentionlabel-update.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Update the properties of a [retentionLabel](../resources/security-retentionlabel.md) object.|
 |[Delete retentionLabel](../api/security-retentionlabel-delete.md)|None|Delete a [retentionLabel](../resources/security-retentionlabel.md) object.|
+|[List filePlanDescriptor](../api/security-retentionlabel-list.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get the filePlanDescriptor resources from the descriptors navigation property.|
```
```suggestion
|[List filePlanDescriptor](../api/security-retentionlabel-list.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get the filePlanDescriptor resources from the descriptors navigation property.|
```
```suggestion
|[List filePlanDescriptor](../api/security-labelsroot-list-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get the filePlanDescriptor resources from the descriptors navigation property.|
```

Danielabom commented on api-reference/beta/resources/security-retentionlabel.md:
```diff
@@ -23,9 +23,12 @@ Represents how customers can manage their data, including whether and for how lo
 |[Add retentionEventType](../api/security-retentioneventtype-post.md)|[microsoft.graph.security.retentionEventType](../resources/security-retentioneventtype.md)|Add eventType by adding the relevant odata property when creating a label.|
 
+
 ## Properties
```
@sseth-msft, please address the comment above I left last week. Thanks!

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List retentionLabels](../api/security-retentionlabel-list.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md) collection|Get a list of the [retentionLabel](../resources/security-retentionlabel.md) objects and their properties.|
```
```suggestion
|[List retentionLabels](../api/security-retentionlabel-list.md)|[microsoft.graph.security.retentionLabel](../resources/security-labelsroot-list-retentionlabel.md) collection|Get a list of the [retentionLabel](../resources/security-retentionlabel.md) objects and their properties.|
```

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List retentionLabels](../api/security-retentionlabel-list.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md) collection|Get a list of the [retentionLabel](../resources/security-retentionlabel.md) objects and their properties.|
+|[Create retentionLabel](../api/security-retentionlabel-post.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Create a new [retentionLabel](../resources/security-retentionlabel.md) object.|
```
```suggestion
|[Create retentionLabel](../api/security-labelsroot-post-retentionlabel.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Create a new [retentionLabel](../resources/security-retentionlabel.md) object.|
```

Danielabom commented on api-reference/beta/resources/security-labelsroot.md:
```diff
@@ -0,0 +1,71 @@
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List retentionLabels](../api/security-retentionlabel-list.md)|[microsoft.graph.security.retentionLabel](../resources/security-labelsroot-list-retentionlabel.md) collection|Get a list of the [retentionLabel](../resources/security-retentionlabel.md) objects and their properties.|
```
```suggestion
|[List retentionLabels](../api/security-labelsroot-list-retentionlabel.md)|[microsoft.graph.security.retentionLabel](../resources/security-labelsroot-list-retentionlabel.md) collection|Get a list of the [retentionLabel](../resources/security-retentionlabel.md) objects and their properties.|
```

Danielabom commented on api-reference/beta/resources/security-subcategorytemplate.md:
```diff
@@ -0,0 +1,61 @@
+|[List subCategories](../api/security-categorytemplate-list-subcategories.md)|[microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) collection|Get a list of the [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) objects and their properties.|
+|[Create subCategories](../api/security-categorytemplate-post-subcategories.md)|[microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md)|Create a new [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) object.|
+|[Get subCategories](../api/security-subcategorytemplate-get.md)|[microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md)|Read the properties and relationships of a [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) object.|
+|[Delete subCategories](../api/security-categorytemplate-post-subcategories.md)|None|Delete a [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) object.|
```
```suggestion
|[Delete subCategories](../api/security-categorytemplate-delete-subcategories.md)|None|Delete a [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) object.|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,78 @@
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List filePlanDescriptors](../api/security-retentionlabel-list.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get a list of the [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) objects and their properties.|
```
```suggestion
|[List filePlanDescriptors](../api/security-labelsroot-list-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get a list of the [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) objects and their properties.|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,78 @@
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List filePlanDescriptors](../api/security-retentionlabel-list.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get a list of the [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) objects and their properties.|
+|[Create filePlanDescriptor](../api/security-retentionlabel-post.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md)|Create a new [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) object.|
```
```suggestion
|[Create filePlanDescriptor](../api/security-labelsroot-post-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md)|Create a new [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) object.|
```

Danielabom commented on api-reference/beta/resources/security-fileplandescriptortemplate.md:
```diff
@@ -0,0 +1,71 @@
+Represents a group of columns to improve the manageability and organization of the content you need to label.
+
+Inherits from [microsoft.graph.entity](../resources/entity.md).
+
```
@sseth-msft, why are we not exposing these methods in the TOC?

sseth-msft replied:
Hi Dani, the file plan descriptor is a collection under which we have the 6 descriptors, they are all individually added to the toc.

Lauragra commented on api-reference/beta/api/security-authoritytemplate-get.md:
```diff
@@ -0,0 +1,91 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of a [microsoft.graph.security.authorityTemplate](../resources/security-authoritytemplate.md) object.
```
Global: We only fully qualify the type name in specific places in the topics. In the summary statement, we just use the resource name without fully qualifying the namespace, for readability. Please update throughout.

sseth-msft replied:
fixed on all method/api files.

Lauragra commented on api-reference/beta/api/security-authoritytemplate-get.md:
```diff
@@ -0,0 +1,91 @@
+
+
+### Response
+The following is an example of the response
```
Global: Please add a period to the end of the sentence in the Response section.

sseth-msft replied:
fixed

Lauragra commented on api-reference/beta/api/security-categorytemplate-delete-subcategories.md:
```diff
@@ -0,0 +1,73 @@
+
+### Response
+The following is an example of the response
+>**Note:** The response object shown here might be shortened for readability.
```
Global: When there's no response body as for delete operations. we don't include this note because the response body is not shortened.

sseth-msft replied:
Removed from all delete operations. All other operations - get, list, post have a response.

Lauragra commented on api-reference/beta/api/security-categorytemplate-list-subcategories.md:
```diff
@@ -0,0 +1,92 @@
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Get the [microsoft.graph.security.subCategoryTemplate](../resources/security-subcategorytemplate.md) resources from the subCategories navigation property.
```
Global: This boilerplate description is not that useful. A more useful description might be something like: "Get a list of subcategories associated with a category template."

sseth-msft replied:
updated as per the suggestion.

Lauragra commented on api-reference/beta/api/security-categorytemplate-list-subcategories.md:
```diff
@@ -0,0 +1,92 @@
+
+## Response
+
+If successful, this method returns a `200 OK` response code and a collection of [subCategoryTemplate](../resources/security-subcategorytemplate.md) objects in the response body.
```
Global: In the Response H2 section, we do use the fully qualified subamespace. Please consistently use the fully qualified name in the Response section (but not in the summary description). See the guidelines for subnamespaces: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211759/OData-subnamespace

sseth-msft replied:
fixed for all APIs/method files.

Lauragra commented on api-reference/beta/api/security-categorytemplate-post-subcategories.md:
```diff
@@ -0,0 +1,104 @@
+
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String|**TODO: Add Description** Inherited from [microsoft.graph.security.filePlanDescriptorTemplate](../resources/security-fileplandescriptor.md). Optional.|
```
Please replace the TODO here with the description.

sseth-msft replied:
added description.

Lauragra commented on api-reference/beta/toc.yml:
```diff
@@ -16118,20 +16118,95 @@ items:
             href: api/security-retentionlabel-update.md
           - name: Delete
             href: api/security-retentionlabel-delete.md
+          - name: List file plan descriptors
```
Why aren't we using "List retention label" and "Create retention label" for the TOC entries here?

sseth-msft replied:
discussed with Laura, OK to remove this and CREATE option under it.

sseth-msft replied:
removed from TOC.

Lauragra commented on api-reference/beta/resources/security-fileplandescriptor.md:
```diff
@@ -0,0 +1,78 @@
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List filePlanDescriptors](../api/security-labelsroot-list-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get a list of the [microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) objects and their properties.|
```
The two methods you're linking to here are List and Create retentionLabels. If there should be List and Create methods on the filePlanDescriptors resource, are we missing those method topics?

sseth-msft replied:
same as above.

Lauragra commented on api-reference/beta/resources/security-retentionlabel.md:
```diff
@@ -18,14 +18,17 @@ Represents how customers can manage their data, including whether and for how lo
 |[Get retentionLabel](../api/security-retentionlabel-get.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Read the properties and relationships of a [retentionLabel](../resources/security-retentionlabel.md) object.|
 |[Update retentionLabel](../api/security-retentionlabel-update.md)|[microsoft.graph.security.retentionLabel](../resources/security-retentionlabel.md)|Update the properties of a [retentionLabel](../resources/security-retentionlabel.md) object.|
 |[Delete retentionLabel](../api/security-retentionlabel-delete.md)|None|Delete a [retentionLabel](../resources/security-retentionlabel.md) object.|
+|[List filePlanDescriptor](../api/security-labelsroot-list-retentionlabel.md)|[microsoft.graph.security.filePlanDescriptor](../resources/security-fileplandescriptor.md) collection|Get the filePlanDescriptor resources from the descriptors navigation property.|
```
In this topic it looks like you're linking to the same methods (list and create retentionLabel) twice, with different method names. Seems like we are missing the files that should go with list and create filePlanDescriptor.

sseth-msft replied:
done.

Lauragra commented on api-reference/beta/toc.yml:
```diff
@@ -16118,20 +16118,95 @@ items:
+          - name: File plan descriptor
+            href: resources/security-fileplandescriptor.md
+            items:
+            - name: File plan reference
```
Should this TOC entry be File plan reference template, to match the resource name?

sseth-msft replied:
check the links to the file, should link to the correct file.

sseth-msft replied:
done
